src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb

```diff
@@ -709,22 +709,28 @@ Namespace Microsoft.VisualStudio.Editors.AppDesCommon
 
         Public Class SerializationProvider
 
+            Private Shared ReadOnly KnownTypes As Type() = {GetType(Size)}
+
             Public Shared Sub Serialize(stream As Stream, value As Object)
-                If value Is Nothing Then
+                If stream Is Nothing Or value Is Nothing Then
```

Comment:
I don't think it makes sense to accept a null stream. I would use `Requires.NotNull` for that parameter.

Reply:
Instead, for null value, consider writing `"<null>"` (or some such string) to the stream in place of the type name. Then, when deserialising, you receive that string then you know to return null.

Reply:
Ah. I looked into it. Since this is a 'drop-in replacement' of BinaryFormatter, I'm going to do the same thing that [Serialize](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.formatters.binary.binaryformatter.serialize?view=net-5.0) and [Deserialize](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.formatters.binary.binaryformatter.deserialize?view=net-5.0) do in these situations. I can use `Requires.NotNull` for most of them. Thanks.